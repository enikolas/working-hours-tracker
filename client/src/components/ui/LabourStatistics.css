.LabourStatistics {
  padding: 20px 20px 6px;
  text-align: right;
  margin: 10px 0 10px 15px;
  width: calc(100% - 15px);
}
