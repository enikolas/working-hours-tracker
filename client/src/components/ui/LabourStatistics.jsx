import React from 'react';
import PropTypes from 'prop-types';

import Gauge from './Gauge';
import BalanceBox from './BalanceBox';

import strings from '../../utils/strings';
import './LabourStatistics.css';

const LabourStatistics = ({
	dayBalance,
	weekBalance,
	contractedTime,
	weekDay,
	totalBalance
}) => (
	<div className="LabourStatistics">
		<Gauge
			description={strings.hoursLabouredOnThisDay}
			value={dayBalance}
			expected={contractedTime}
		/>
		<Gauge
			description={strings.hoursBalanceOnWeekUpToNow}
			value={weekBalance}
			expected={contractedTime * weekDay}
		/>
		<BalanceBox
			description={strings.hoursBalanceForToday}
			value={totalBalance}
		/>
	</div>
);

LabourStatistics.propTypes = {
	totalBalance: PropTypes.number,
	contractedTime: PropTypes.number,
	dayBalance: PropTypes.number,
	weekBalance: PropTypes.number,
	weekDay: PropTypes.number.isRequired
};

LabourStatistics.defaultProps = {
	totalBalance: 0,
	contractedTime: 8,
	dayBalance: 0,
	weekBalance: 0
};

export default LabourStatistics;
