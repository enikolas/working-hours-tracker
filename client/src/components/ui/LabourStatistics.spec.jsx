import React from 'react';
import { shallow } from 'enzyme';

import LabourStatistics from './LabourStatistics';

const defaultProps = {
	totalBalance: 0,
	contractedTime: 8 * 60,
	dayBalance: 8 * 60,
	weekBalance: 8 * 60 * 2,
	weekDay: 3
};

describe('LabourStatistics', () => {
	describe('render', () => {
		it('should properly render with surplus or debit', () => {
			const props = { ...defaultProps, weekBalance: 8 * 60 * 4 };
			const wrapperSurplus = shallow(<LabourStatistics {...props} />);
			const wrapperDebt = shallow(<LabourStatistics {...defaultProps} />);

			expect(wrapperDebt).toMatchSnapshot();
			expect(wrapperSurplus).toMatchSnapshot();
		});
	});
});
