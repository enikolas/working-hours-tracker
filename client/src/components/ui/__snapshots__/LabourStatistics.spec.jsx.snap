// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabourStatistics render should properly render with surplus or debit 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LabourStatistics
    contractedTime={480}
    dayBalance={480}
    totalBalance={0}
    weekBalance={960}
    weekDay={3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Gauge
          description="Horas trabalhadas no dia:"
          expected={480}
          value={480}
        />,
        <Gauge
          description="Na semana até esse dia"
          expected={1440}
          value={960}
        />,
        <BalanceBox
          description="Balanço de horas atual:"
          value={0}
        />,
      ],
      "className": "LabourStatistics",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Horas trabalhadas no dia:",
          "expected": 480,
          "value": 480,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Na semana até esse dia",
          "expected": 1440,
          "value": 960,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Balanço de horas atual:",
          "value": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Gauge
            description="Horas trabalhadas no dia:"
            expected={480}
            value={480}
          />,
          <Gauge
            description="Na semana até esse dia"
            expected={1440}
            value={960}
          />,
          <BalanceBox
            description="Balanço de horas atual:"
            value={0}
          />,
        ],
        "className": "LabourStatistics",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Horas trabalhadas no dia:",
            "expected": 480,
            "value": 480,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Na semana até esse dia",
            "expected": 1440,
            "value": 960,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Balanço de horas atual:",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`LabourStatistics render should properly render with surplus or debit 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LabourStatistics
    contractedTime={480}
    dayBalance={480}
    totalBalance={0}
    weekBalance={1920}
    weekDay={3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Gauge
          description="Horas trabalhadas no dia:"
          expected={480}
          value={480}
        />,
        <Gauge
          description="Na semana até esse dia"
          expected={1440}
          value={1920}
        />,
        <BalanceBox
          description="Balanço de horas atual:"
          value={0}
        />,
      ],
      "className": "LabourStatistics",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Horas trabalhadas no dia:",
          "expected": 480,
          "value": 480,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Na semana até esse dia",
          "expected": 1440,
          "value": 1920,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Balanço de horas atual:",
          "value": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Gauge
            description="Horas trabalhadas no dia:"
            expected={480}
            value={480}
          />,
          <Gauge
            description="Na semana até esse dia"
            expected={1440}
            value={1920}
          />,
          <BalanceBox
            description="Balanço de horas atual:"
            value={0}
          />,
        ],
        "className": "LabourStatistics",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Horas trabalhadas no dia:",
            "expected": 480,
            "value": 480,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Na semana até esse dia",
            "expected": 1440,
            "value": 1920,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Balanço de horas atual:",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
